### ISSUE TITLE
How to extend action with several methods using OpenApiViewExtension?

### ISSUE THREAD

**Issue by MatejMijoski** [tfranzel/drf-spectacular#613]
I currently have an action on a view that support GET, POST and DELETE. How can I extend their schema by using the OpenApiViewExtensionClass?

```
class TestAPISchema(OpenApiViewExtension):
    target_class = "test.views.TestAPI"

    def view_replacement(self):
        @extend_schema_view(
            test_action=extend_schema(
                 ....
                some params for GET Method
                 ....
            ),
            test_action=extend_schema(
                 ....
                some params for POSTMethod
                 ....
            ),
         )
```
I can't add the same action twice because it's a dictionary. I'm not sure how to resolve this.

**Comment by tfranzel** [comment]
Hi, this is a limitation of the `@extend_schema_view` convenience decorator.

To make this work you would need to split the action into one action function per http method:

https://www.django-rest-framework.org/api-guide/viewsets/#routing-additional-http-methods-for-extra-actions

then you could target `test_action` and e.g. `test_action_post` individually, if you would do:


```python
@action(methods=['GET'])
def test_action():
   ....

@test_action.mapping.post
def test_action_post():
   ....
```

**Comment by tfranzel** [comment]
this has been addressed in #648. `extend_schema_view` now supports scoped lists
